app/services/locale_ninja/github_api_service.rb

```diff
@@ -19,22 +19,24 @@ def translation_branch(branch)
       branch.ends_with?('__translations') ? branch : "#{branch}__translations"
     end
 
-    def locale_files_path(dir = 'config/locales', branch: 'translations')
+    def locale_files(dir = 'config/locales', branch: 'translations')
       branch = translation_branch(branch) if branch?(translation_branch(branch))
-      @client.contents(repository_fullname, path: dir, ref: "heads/#{branch}").map do |file|
-        if file.type == 'dir'
-          locale_files_path(file.path, branch:)
-        else
-          file.path
-        end
-      end.flatten
+      @client.tree(repository_fullname, "heads/#{branch}", recursive: true).tree.select do |file|
```

Comment:
(rien testé mais) on devrait pouvoir récupérer directement le sha du dossier pour pas récupérer tout le reste

Reply:
It doesn't seem possible to retrieve the local folder sha beforehand, to avoid having to retrieve the entire tree.

Reply:
I mean, it is, but the method would be called twice, which isn't necessary.

Reply:
wouldn't be possible to cache it the first time ? does the sha change when we push ?

Reply:
You can try it if you want, but it doesn't seem to improve system performance significantly.
If anything, the opposite is true.

The main concerns is probably to cache yaml files and to refresh cache when an update event occurs on the repo as said here : #98 

By the way, the tree method of Octokit::Client is pretty fast.

Reply:
tried and found nothing conclusive